sdk/devcenter/azure-developer-devcenter/src/main/java/com/azure/developer/devcenter/EnvironmentsClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+        return this;
+    }
+
+    /*
+     * The retry policy that will attempt to retry failed requests, if applicable.
+     */
+    @Generated private RetryPolicy retryPolicy;
+
+    /**
+     * Sets The retry policy that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the EnvironmentsClientBuilder.
+     */
+    @Generated
+    public EnvironmentsClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Builds an instance of DevCenterClientImpl with the provided parameters.
+     *
+     * @return an instance of DevCenterClientImpl.
+     */
+    @Generated
+    private DevCenterClientImpl buildInnerClient() {
+        HttpPipeline localPipeline = (pipeline != null) ? pipeline : createHttpPipeline();
+        String localDevCenterDnsSuffix = (devCenterDnsSuffix != null) ? devCenterDnsSuffix : "devcenter.azure.com";
+        DevCenterServiceVersion localServiceVersion =
+                (serviceVersion != null) ? serviceVersion : DevCenterServiceVersion.getLatest();
+        DevCenterClientImpl client =
+                new DevCenterClientImpl(
+                        localPipeline,
+                        JacksonAdapter.createDefaultSerializerAdapter(),
+                        tenantId,
+                        devCenter,
+                        localDevCenterDnsSuffix,
+                        localServiceVersion);
+        return client;
+    }
+
+    @Generated
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+                (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        if (clientOptions == null) {
+            clientOptions = new ClientOptions();
+        }
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
+        String applicationId = CoreUtils.getApplicationId(clientOptions, httpLogOptions);
+        policies.add(new UserAgentPolicy(applicationId, clientName, clientVersion, buildConfiguration));
+        policies.add(new RequestIdPolicy());
+        policies.add(new AddHeadersFromContextPolicy());
+        HttpHeaders headers = new HttpHeaders();
+        clientOptions.getHeaders().forEach(header -> headers.set(header.getName(), header.getValue()));
+        if (headers.getSize() > 0) {
+            policies.add(new AddHeadersPolicy(headers));
+        }
+        policies.addAll(
+                this.pipelinePolicies.stream()
+                        .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_CALL)
+                        .collect(Collectors.toList()));
```

Comment:
For @weidongxu-microsoft, minor nit as I've seen this elsewhere in generated code, we can flip this to something more performant. Right now, this will create a List to be used by `addAll` which many times converts the list back to an array. In the builder this won't be too bad as there is lower usage and it's not a hot path but in other places it could be:

```java
this.pipelinePolicies().stream()
    .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_CALL)
    .forEach(p -> policies.add(p));
```